tests/server/sockfilt.c

```diff
@@ -786,90 +786,89 @@ static int select_ws(int nfds, fd_set *readfds, fd_set *writefds,
   }
 
   /* wait for one of the internal handles to trigger */
-  wait = WaitForMultipleObjectsEx(nfd, handles, FALSE, milliseconds, FALSE);
+  wait = WaitForMultipleObjectsEx(nfd, handles, FALSE, timeout_ms, FALSE);
 
   /* wait for internal mutex to lock event handling in threads */
   WaitForSingleObjectEx(mutex, INFINITE, FALSE);
 
   /* loop over the internal handles returned in the descriptors */
-  for(idx = 0; idx < nfd; idx++) {
-    curl_socket_t sock = data[idx].fd;
-    handle = handles[idx];
-    fds = curlx_sktosi(sock);
+  ret = 0; /* number of ready file descriptors */
+  for(i = 0; i < nfd; i++) {
+    fd = data[i].fd;
+    handle = handles[i];
+    wsasock = curlx_sitosk(fd);
 
     /* check if the current internal handle was triggered */
-    if(wait != WAIT_FAILED && (wait - WAIT_OBJECT_0) <= idx &&
+    if(wait != WAIT_FAILED && (wait - WAIT_OBJECT_0) <= i &&
```

Comment:
This PR looks fine but I think you may need better checking here. From what I've read the return code only represents the number of handles when it's in a specific range. _WAIT_OBJECT_0 to (WAIT_OBJECT_0 + nCount– 1)_

https://docs.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-waitformultipleobjectsex#return-value

Reply:
I am not exactly sure I can follow your comment. Do you mean that the code is missing a check for `wait >= WAIT_OBJECT_0 ` or `(wait - WAIT_OBJECT_0) >= 0`? The main check is basically `wait != WAIT_FAILED` to check for an error condition first and then also use `WaitForSingleObjectEx` again to check the individual handles, so the existing `(wait - WAIT_OBJECT_0) <= i` in between is only done to not call `WaitForSingleObjectEx` for handles that we know are not ready. But sure, this can be improved by checking the lower bounds as well.

Reply:
I will address this in a separate PR and will now go ahead and merge this.

Reply:
I changed the condition in the following way, but now the compilers are complaining that the new checks are unnecessary, because `WAIT_OBJECT_0` is 0 and `DWORD` is unsigned. See #5260:
https://github.com/curl/curl/blob/749cfaed84c2a88b3d998413e13dcc8b0769ce9c/tests/server/sockfilt.c#L812-L814